app/models/katello/content_view_version.rb

```diff
@@ -247,8 +247,17 @@ def package_groups
       PackageGroup.in_repositories(archived_repos).uniq
     end
 
-    def check_ready_to_promote!
+    def check_ready_to_promote!(to_env)
       fail _("Default content view versions cannot be promoted") if default?
+
+      unless content_view.composite_action_allowed?(to_env)
+        fail _("This composite view cannot be promoted until all of the component content view versions "\
+            "have been promoted to %{env}.  This restriction is optional and can be modified in the "\
+            "Administrator -> Settings page using the restrict_composite_view flag.") %
+            { :env => to_env.name }
```

Comment:
Should this message/condition be somehow centralized into the content_view model to DRY it?